designs/generic-git-provider-gitops.md

```diff
@@ -0,0 +1,205 @@
+more feature rich, provider-specific git integrations (e.g. Gitlab provider which creates/cleans up the repo at the Gitlab API level) in the future as user demand emerges.
+
+## Goals and Objectives
+As an EKS Anywhere user, I want to be able to use a generic git repository with SSH authentication as the source of truth for Flux.
+
+The goal of this design is to unlock the ability for EKS-A users to use any git provider which supports generic authentication methods like SSH authentication as their GitOps provider.
+
+## Tenants
+- Supports any generic generic git providers with password/SSH auth
+- Minimal -- simple and straight forward
+- No tight coupling preventing abstraction of Flux configuration into a curated package in the future
+
+## Scope
+### In-Scope
+- addition of a generic git provider
+- SSH and password auth for generic git provider
+
+### Out-of-Scope
+- provider-specific actions (create, delete repository) for aribitrary git providers
+- refactoring to use git executable vs gogit library
+- refactoring to use flux git provider Go libraries
+
+## Overview of Solution
+Currently, EKS Anywhere uses a [Github specific Flux bootstrap](https://fluxcd.io/docs/cmd/flux_bootstrap_github/) command to install and bootstrap flux on a new EKS Anywhere cluster (`flux bootstrap github`) and uses the Github API to manage the repository itself (create it during `create cluster`, delete it during `delete cluster`).
+
+In order to support the broadest array of git hosting solutions via basic authentication like SSH and password/username, the generic git provider will simplify this process, accepting basic authentication parameters and passing them to Flux's [`flux bootstrap git`](https://fluxcd.io/docs/cmd/flux_bootstrap_git/) command, bypassing the repository management steps we would take for a specific provider. This will enable users who rely on a variety of Git providers to use GitOps with EKS Anywhere right away.
+
+## User Experience
+
+### Create
+The EKS Anywhere CLI user will provide the configuration neccessary to authenticate to their generic git repository via a `git` provider in the `gitOpsConfig` spec. The changes to this spec are discussed in more detail below.
+
+```yaml
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: GitOpsConfig
+metadata:
+  name: my-generic-git-provider-gitops
+spec:
+  flux:
+    fluxSystemNamespace: "some-other-namespace"
+    git:
+      repositoryUrl: git@github.com:danbudris/design-doc-cluster.git
+      username: danbudris
+      privateKeyFile: '/path/to/my/private_key.rsa'
```

Comment:
I don't think we should have paths defined in the spec. We should define it as an env var and store as a secret once it gets applied if we need it for upgrades and later actions.

Reply:
agreed, will update.